File: js/3.js
// What is the largest prime factor of 600851475143?

factorise = function(num) {
	var factors = [];
	var dvs = 2; // the smallest prime
	var brkpt = Math.sqrt(num); // the largest possible factor
	while (num > 1) { 
		while (num % dvs == 0) {
			factors.push(dvs);
			num /= dvs; 
		}
		dvs++; 
		// we will always hit a prime before we hit a composite
		// and each primes remove all composites it is a factor of
	}
	return factors;
};

var factors = factorise(600851475143);
// since we started with the lowest factors,
// the highest is at the end

console.log(factors[factors.length-1]); // 6857

File: c/3.c
// What is the largest prime factor of the number 600851475143

#include <stdio.h>

int factorise(long n);

int main() 
{
    int res;
    long num = 600851475143;

    res = factorise(num);
    printf("%d\n", res);
    return 0;
}

int factorise(long n) {
    int div, factor;

    div = 2; // smallest prime

    while (n > 1)
    {
        while (n % div == 0)
        {
            factor = div; // last is expected to be largest...
            n = n / div; 
        }
        div++; // ... because we are strictly incrementing
    }
    return factor; // 6857
}